staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -190,14 +190,18 @@ type LeaderElector struct {
 func (le *LeaderElector) Run(ctx context.Context) {
 	defer func() {
 		runtime.HandleCrash()
-		le.config.Callbacks.OnStoppedLeading()
+		if le.config.Callbacks.OnStoppedLeading != nil {
+			go le.config.Callbacks.OnStoppedLeading()
```

Comment:
Probably not a good idea to switch this to a new goroutine without serious investigation. Our general behavior in things like the KCM to losing leader election is to kill our own process. Moving this to a goroutine means that the callback is now racing to act prior to the process being killed.

Reply:
That makes sense. I re-uped the commit and removed the goroutine.